src/parser/parser.ts

```diff
@@ -113,6 +114,9 @@ export function parseAt(source: string, num: number) {
 }
 
 export function parse(source: string, context: Context) {
+  if (isFullJSChapter(context.chapter)) {
+    return parseFullJS(source, context)
+  }
```

Comment:
Will `fullJS` be calling `parse` in the frontend? 

cmmiw I don't think we will need it since everything goes through `runInContext` and we will not be doing any validating/parsing before execution.